app/src/main/java/lu/geoportail/map/LuxTileServer.kt

```diff
@@ -0,0 +1,149 @@
+package lu.geoportail.map
+
+import android.content.Context
+import android.content.res.Resources
+import android.database.sqlite.SQLiteDatabase
+import com.koushikdutta.async.ByteBufferList
+import com.koushikdutta.async.http.WebSocket
+import com.koushikdutta.async.http.server.AsyncHttpServer
+import com.koushikdutta.async.http.server.HttpServerRequestCallback
+import com.koushikdutta.async.http.server.AsyncHttpServerRequest
+import com.koushikdutta.async.http.server.AsyncHttpServerResponse
+import java.io.ByteArrayInputStream
+import java.io.File
+import java.nio.ByteBuffer
+import java.nio.file.Files
+import java.util.ArrayList
+
+class LuxTileServer (context:Context, resources:Resources) {
+    private var db: SQLiteDatabase? = null
```

Comment:
I'm tented to use a `lateinit`property to this database object. It simplifies the code (no null check), and compiler check if the resource is used before being init.